components/BirthdayList.js

```diff
@@ -1,18 +1,106 @@
 import styled from "styled-components";
+import { useState } from "react";
+
+export default function BirthdayList({
+  id,
+  name,
+  birthday,
+  ideas,
+  onUpdateEntry,
+  onDelete,
+}) {
+  const [edit, setEdit] = useState(false);
+
+  function handleEditSubmit(event) {
+    event.preventDefault();
+    const adaptedName = event.target.adaptedName.value;
+    const adaptedIdeas = event.target.adaptedIdeas.value;
+    const adaptedBirthday = event.target.adaptedBirthday.value;
+    const editedEntry = {
+      id,
+      name: adaptedName,
+      birthday: adaptedBirthday,
+      ideas: adaptedIdeas,
+    };
+    onUpdateEntry(editedEntry);
+    setEdit(!edit);
+  }
 
-export default function BirthdayList({ id, name, birthday, ideas }) {
   return (
-    <StyledEntry>
-      <div>{name}</div>
-      <div>{ideas.join(", ")}</div>
-      <StyledBirthday>{birthday}</StyledBirthday>
-    </StyledEntry>
+    <>
+      {edit === true ? (
+        <StyledEditForm onSubmit={handleEditSubmit}>
+          <StyledAdaptedName
+            name="adaptedName"
+            placeholder="Namen anpassen..."
+            required
+          ></StyledAdaptedName>
+          <StyledAdaptedIdeas
+            name="adaptedIdeas"
+            placeholder="Ideen anpassen..."
+          ></StyledAdaptedIdeas>
+          <StyledAdaptedBirthday
+            name="adaptedBirthday"
+            placeholder="Geburtstag anpassen..."
+            required
+          ></StyledAdaptedBirthday>
```

Comment:
das Feld brauchst du nicht, wenn du in Zeile 53 den content als editable deklarierst, oder? Falls du es drinlassen willst, würde ich das Format hier genauso übernehmen, wie im ursprünglichen Formular, sonst kann man seltsame Formate eingeben